File: dummy_creator.cpp
// includes
#include "dummy_creator.h"
#include "generic_protocol/dummy_creator.h"
#include "basic_parser/dummy_creator.h"
#include "basic_objects/dummy_creator.h"
#include "object_initializer.h"

namespace user_management_protocol
{

namespace dummy
{

// enums

gender_e create__gender_e()
{
    static const unsigned SIZE = 3;

    static const gender_e values[SIZE] = { gender_e::UNDEF, gender_e::MALE, gender_e::FEMALE,  };

    auto res = values[ ::basic_parser::dummy::create__uint32() % SIZE ];

    return res;
}

// objects

UserInfo create__UserInfo()
{
    UserInfo res;

    ::user_management_protocol::initialize( & res
        , dummy::create__gender_e()
        , ::basic_parser::dummy::create__string()
        , ::basic_parser::dummy::create__string()
        , ::basic_parser::dummy::create__string()
        , ::basic_parser::dummy::create__string()
        , ::basic_parser::dummy::create__string()
        , ::basic_parser::dummy::create__string()
        , ::basic_parser::dummy::create__string()
        , ::basic_parser::dummy::create__string()
        );

    return res;
}

// messages

SetUserInfoRequest * create__SetUserInfoRequest()
{
    auto res = new SetUserInfoRequest;

    ::user_management_protocol::initialize( res
        , ::basic_parser::dummy::create__uint32()
        , dummy::create__UserInfo()
        );

    return res;
}

SetUserInfoResponse * create__SetUserInfoResponse()
{
    auto res = new SetUserInfoResponse;

    ::user_management_protocol::initialize( res
        );

    return res;
}

GetUserInfoRequest * create__GetUserInfoRequest()
{
    auto res = new GetUserInfoRequest;

    ::user_management_protocol::initialize( res
        , ::basic_parser::dummy::create__uint32()
        );

    return res;
}

GetUserInfoResponse * create__GetUserInfoResponse()
{
    auto res = new GetUserInfoResponse;

    ::user_management_protocol::initialize( res
        , ::basic_parser::dummy::create__uint32()
        , dummy::create__gender_e()
        , ::basic_parser::dummy::create__string()
        , ::basic_parser::dummy::create__string()
        , ::basic_parser::dummy::create__string()
        , ::basic_parser::dummy::create__string()
        , ::basic_parser::dummy::create__string()
        , ::basic_parser::dummy::create__string()
        , ::basic_parser::dummy::create__string()
        , ::basic_parser::dummy::create__string()
        );

    return res;
}

} // namespace dummy

} // namespace user_management_protocol

File: dummy_creator.php
<?php

namespace user_management_protocol;


// base include
require_once __DIR__.'/../generic_protocol/dummy_creator.php';
// includes
require_once __DIR__.'/../basic_parser/dummy_creator.php';
require_once __DIR__.'/../basic_objects/dummy_creator.php';

// own includes
require_once __DIR__.'/../user_management_protocol/object_initializer.php';

// enums

function create_dummy__gender_e()
{
    $SIZE = 3;

    $values = array( gender_e__UNDEF, gender_e__MALE, gender_e__FEMALE,  );

    $res = $values[ \basic_parser\create_dummy__int32() % $SIZE ];

    return $res;
}

// objects

function create_dummy__UserInfo()
{
    $res = new UserInfo;

    initialize__UserInfo( $res
        , create_dummy__gender_e()
        , \basic_parser\create_dummy__string()
        , \basic_parser\create_dummy__string()
        , \basic_parser\create_dummy__string()
        , \basic_parser\create_dummy__string()
        , \basic_parser\create_dummy__string()
        , \basic_parser\create_dummy__string()
        , \basic_parser\create_dummy__string()
        , \basic_parser\create_dummy__string()
        );
    return $res;
}

// messages

function create_dummy__SetUserInfoRequest()
{
    $res = new SetUserInfoRequest;

    initialize__SetUserInfoRequest( $res
        , \basic_parser\create_dummy__int32()
        , create_dummy__UserInfo()
        );
    return $res;
}

function create_dummy__SetUserInfoResponse()
{
    $res = new SetUserInfoResponse;

    initialize__SetUserInfoResponse( $res
        );
    return $res;
}

function create_dummy__GetUserInfoRequest()
{
    $res = new GetUserInfoRequest;

    initialize__GetUserInfoRequest( $res
        , \basic_parser\create_dummy__int32()
        );
    return $res;
}

function create_dummy__GetUserInfoResponse()
{
    $res = new GetUserInfoResponse;

    initialize__GetUserInfoResponse( $res
        , \basic_parser\create_dummy__int32()
        , create_dummy__gender_e()
        , \basic_parser\create_dummy__string()
        , \basic_parser\create_dummy__string()
        , \basic_parser\create_dummy__string()
        , \basic_parser\create_dummy__string()
        , \basic_parser\create_dummy__string()
        , \basic_parser\create_dummy__string()
        , \basic_parser\create_dummy__string()
        , \basic_parser\create_dummy__string()
        );
    return $res;
}

// namespace_end user_management_protocol


?>
